### PR TITLE
chore: repository review

### PR DESCRIPTION
**Pull request by dan-online**

### CODE REVIEW

File: src/lib/structures/Josh.ts [hunk 1]
```diff
@@ -639,7 +674,12 @@ export class Josh<StoredValue = unknown> {
 
     this.runBehaviorOnPayloadError(payload);
 
-    if (isPayloadWithData<Record<string, StoredValue>>(payload)) return this.convertBulkData(payload.data, returnBulkType);
+    if (isPayloadWithData<Record<string, StoredValue>>(payload)) {
+      return this.convertBulkData(
+        payload.data,
+        isPrimitive(valueOrReturnBulkType) || typeof valueOrReturnBulkType === 'symbol' ? returnBulkType : valueOrReturnBulkType || returnBulkType
```
Comment (RealShadowNova) [inline src/lib/structures/Josh.ts:680]: bulk is not a symbol

**Review by RealShadowNova — CHANGES_REQUESTED** [review]

File: src/lib/structures/Josh.ts (hunk 1 above)
Comment (dan-online) [reply src/lib/structures/Josh.ts:680]: Hence why we check, and if so, don't use valueOrReturnBulkType

Comment (RealShadowNova) [reply src/lib/structures/Josh.ts:680]: Why would someone pass a symbol? Where in the typings does this method allow for one?

File: src/lib/structures/Josh.ts [hunk 2]
```diff
@@ -595,36 +623,43 @@ export class Josh<StoredValue = unknown> {
...
   public async filter<BulkType extends keyof ReturnBulk<StoredValue>>(
     hook: Payload.Hook<StoredValue>,
-    _value?: undefined,
     returnBulkType?: BulkType
   ): Promise<ReturnBulk<StoredValue>[BulkType]>;
 
   public async filter<BulkType extends keyof ReturnBulk<StoredValue>>(
     pathOrHook: Path | Payload.Hook<StoredValue>,
-    value?: Primitive,
+    valueOrReturnBulkType?: Primitive | BulkType,
```
Comment (RealShadowNova) [inline src/lib/structures/Josh.ts:636]: Changes made for this method should be reverted.

Comment (dan-online) [reply src/lib/structures/Josh.ts]: Removed

**Review by RealShadowNova — APPROVED** [review]

**Review by RealShadowNova — APPROVED** [review]

**Pull request merged**